validate/validate.h

```diff
@@ -128,6 +129,11 @@ static inline bool IsHostname(const string& to_validate) {
   return true;
 }
 
+static inline size_t Utf8Len(const string& narrow_string) {
+  std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
+  return converter.from_bytes(narrow_string).size();
```

Comment:
It looks like this can be done without converting the string, just using `std::codecvt_utf8::length`.

Reply:
Actually @akonradi looks like std::codecvt_utf8::length requires specifying a locale:

- https://en.cppreference.com/w/cpp/locale/codecvt/length

which is what we don't want because we might not have the locale installed (for example our docker image fails to find: "en_US.UTF-8")

Reply:
The example uses a locale but that's not necessary. It doesn't work, though, because `std::codecvt::length` returns the number of input characters consumed, not the number of output characters. I suspect it also just allocates the same string internally, so not really a performance win.